packages/app/src/SearchPage.tsx

```diff
@@ -461,6 +461,13 @@ function SearchPage() {
   );
 
   const [showSaveSearchModal, setShowSaveSearchModal] = useState(false);
+  const [saveSearchModalData, setSaveSearchModalData] = useState<{
```

Comment:
I think this can be combined with `showSaveSearchModal`, maybe can be renamed into something like `saveSearchModalMode` and then just have `'update' | 'save' | 'hidden'` or something like that.